src/picongpu/include/plugins/radiation/utilities.hpp

```diff
@@ -50,6 +50,18 @@ namespace util
         return a * a*a;
     }
 
+    template<typename A> /// a generic power-4 function
+    HDINLINE A quad(A a)
+    {
+        return a*a * a*a;
+    }
+
+    template<typename A, typename R> /// a more generic power-4 function
```

Comment:
where is this useful?

even the struct version below, which is not instrumenting the operator but the class, is actually a "weird" way of creating an implicit cast from an explicit class.